projects/RabbitMQ.Client/client/impl/AsyncConsumerWorkService.cs

```diff
@@ -49,7 +49,10 @@ public WorkPool(ModelBase model)
 
             public void Start()
             {
-                _worker = Task.Run(Loop, CancellationToken.None);
+                _worker = Task.Run(Loop, _tokenSource.Token)
+                // continuation ignores TaskCancelledException
+                // (the Stop had been called before Loop started execution) 
+                    .ContinueWith((originalTask) => {}, TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
The above code _seems_ fine to me. Does the `.ContinueWith` ensure the `TaskCancelledException` is raised?

Reply:
No, on the contrary, this ensures that `TaskCancelledException` is ignored because the current code silently ignores the `TaskCancelledException` in the body of the `Loop` method and I thought I should not break the public behavior. This continuation ignores the state of the original task all together, but can be changed if need arises to something around the lines of:
```csharp
(originalTask) =>
{
    if (originalTask.IsCanceled)
    { /* do nothing */ }
    else if (originalTask.IsFaulted)
    { ExceptionDispatchInfo.Capture(originalTask.Exception).Throw(); }
}
```
But the inner code never throws in the current version as far as I am concerned.